synapse/handlers/profile.py

```diff
@@ -307,6 +317,63 @@ async def set_avatar_url(
 
         await self._update_join_states(requester, target_user)
 
+    @cached()
```

Comment:
Not a hug deal, but do we expect this to be called often enough with the same `mxc` that we should cache it?

Reply:
Yes, because this is called both when setting your global profile and when setting a per-room one. Except setting a global profile then updates your per-room profile in every room you're in. So if I'm in hundreds of rooms then a single global avatar change is going to trigger this function to be called hundreds of times.

Reply:
Right! Pretty much trying to make #1297 not worse!

Reply:
Yep exactly!